### DISCUSSION TITLE
can I run the airflow command remotely?

### DISCUSSION THREAD

**Discussion by jinzishuai** [apache/airflow#16361] [Q&A]
Hi there,

I have an airflow system deployed via the helm chart from https://github.com/astronomer/airflow-chart in a k8s cluster. Its version is
```
Astronomer Certified 1.10.10.post4 based on Apache Airflow 1.10.10
```

I have been running the `airflow backfill` command to run backfills on the `airflow-scheduler` pod.
 
I have two questions
1. Can I run the backfill command on the `airflow-webserver` pod? Basic tests seem to suggest OK.
2. Can I install the airflow command externally and run it remotely? If possible, what is it talking to? Does the airflow command use the API service from the webserver or does it directly communicate with the database? If so, what kind of client configuration do I need?  There is a brief section in https://airflow.apache.org/docs/apache-airflow/1.10.10/usage-cli.html talking about `[cli]` in the `airflow.cfg` but it is unclear to me how it authenticates. I can see the default values allows local communication but I wonder if I change it to something like `endpoint_url = https://airflow.example.com`, would it work at all? 
![image](https://user-images.githubusercontent.com/1074685/121456984-29cac800-c964-11eb-9dd5-290ee8b490d9.png)

Thank you very much.

**Comment by jedcunningham** [comment]
Hey @jinzishuai,

Basically you need to run it from your 'normal' Airflow environment (DAGs, config, etc). Doing it from the webserver pod in 1.10.x should be fine, but some commands need the DAGs which wouldn't be there in 2.x. Generally, the scheduler pod is a good, "should always work" place.

The cli does communicate with the db directly, so you can get it to work remotely if you configure the instance properly (e.g. same db, DAGs, config). Some subcommands work remotely in theory, but I think in practice it is pretty sparse feature wise.

If you aren't aware, 1.10.x reaches EOL tomorrow(!!), so you should probably plan to upgrade sooner rather than later. 2.0+ has a much more featureful rest api, depending on what you are looking to do.

**Reply by jinzishuai** [reply]
thank you @jedcunningham   

I actually got it working on the webserver pod. Needs to do some k8s serviceaccount configuration in order to get the k8s executor to work but the local backfill option with `-l` works without that. 

Yes, we are planning on the move to airflow-2.0.  

Just curious, in airflow-2, the webserver won't have the DAG any more (due to dag serialization from what I read). Would backfill still work from the webserver pod in airflow-2?

**Reply by jedcunningham** [reply]
Backfill does require the DAG to be present so it won't work from the webserver (assuming the DAGs aren't present in the pod, of course).
